crates/vm/levm/src/memory.rs

```diff
@@ -47,49 +49,58 @@ pub fn load_range(memory: &mut Memory, offset: usize, size: usize) -> Result<&[u
         .ok_or(VMError::OutOfOffset)
 }
 
-pub fn try_store_word(memory: &mut Memory, offset: usize, word: U256) -> Result<(), VMError> {
-    try_resize(
-        memory,
-        offset
-            .checked_add(WORD_SIZE_IN_BYTES_USIZE)
-            .ok_or(VMError::OutOfOffset)?,
-    )?;
+pub fn try_store_word(memory: &mut Memory, offset: U256, word: U256) -> Result<(), VMError> {
+    let new_size: usize = offset
+        .checked_add(WORD_SIZE_IN_BYTES_USIZE.into())
+        .ok_or(VMError::OutOfOffset)?
+        .try_into()
+        .map_err(|_err| VMError::VeryLargeNumber)?;
+
+    try_resize(memory, new_size)?;
     let mut word_bytes = [0u8; WORD_SIZE_IN_BYTES_USIZE];
     word.to_big_endian(&mut word_bytes);
     try_store(memory, &word_bytes, offset, WORD_SIZE_IN_BYTES_USIZE)
 }
 
-pub fn try_store_data(memory: &mut Memory, offset: usize, data: &[u8]) -> Result<(), VMError> {
-    try_resize(
-        memory,
-        offset.checked_add(data.len()).ok_or(VMError::OutOfOffset)?,
-    )?;
+pub fn try_store_data(memory: &mut Memory, offset: U256, data: &[u8]) -> Result<(), VMError> {
+    let new_size = offset
+        .checked_add(data.len().into())
+        .ok_or(VMError::OutOfOffset)?
+        .try_into()
+        .map_err(|_err| VMError::VeryLargeNumber)?;
+    try_resize(memory, new_size)?;
     try_store(memory, data, offset, data.len())
 }
 
 pub fn try_store_range(
     memory: &mut Memory,
-    offset: usize,
+    offset: U256,
     size: usize,
     data: &[u8],
 ) -> Result<(), VMError> {
-    try_resize(
-        memory,
-        offset.checked_add(size).ok_or(VMError::OutOfOffset)?,
-    )?;
+    let new_size = offset
```

Comment:
I think that here you need to check if the size is equal to zero

Reply:
Done in [3d1abad](https://github.com/lambdaclass/ethrex/pull/1458/commits/3d1abadc3833b3d05ef2587145061b49f3c8e3d7)